File: 프로그래머스/lv2/42578. 의상/의상.java
import java.util.*;
class Solution {
    static int answer; 
    public int solution(String[][] clothes) {
        Map <String, Integer> map = new HashMap<>();
        for(int i=0; i<clothes.length; i++){
            map.put(clothes[i][1],map.getOrDefault(clothes[i][1],0)+1);
        }
        answer = 1;
        map.forEach((key,value) ->{
            answer *= (value+1);
        });
        
        
        return answer-1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    dic = {}
    for cloth,kind in clothes:
        if kind in dic:
            dic[kind]+=[cloth]
            continue         
        dic[kind]=[cloth]

    answer = 1
    for kind in dic:
        answer*=(len(dic[kind])+1)
    return answer-1